### PR TITLE
FIX Don't use the deprecated 'none' encryptor

### PR DESCRIPTION
**Pull request by GuySartorelli**
Setting a Member's `PasswordEncryption` to "none" ends up using the deprecated `PasswordEncryptor_None` class.

There's no reason to not use encryption here.

## Issue
- https://github.com/silverstripe/silverstripe-behat-extension/issues/271

### CODE REVIEW

**Review by emteknetnz — CHANGES_REQUESTED** [review]
You'll need to run this one across sink .. there may actually be a reason we don't know about

**Comment by GuySartorelli** [comment]
Sink seems excessive, I'd have thought installer maybe. But to avoid ping pong I'll just do it with all of sink. Link to CI run will be in the linked issue description.

File: src/Context/LoginContext.php [hunk 1]
```diff
@@ -315,14 +316,21 @@ protected function generateMember($email, $password, $group, $identifier)
 
         // make sure any validation for password is skipped, since we're not testing complexity here
         $validator = Member::password_validator();
-        Member::set_password_validator(null);
+        $nullValidator = new PasswordValidator();
+        $nullValidator->setMinLength(0);
+        $nullValidator->setMinTestScore(0);
+        $nullValidator->setHistoricCount(0);
+        Member::set_password_validator($nullValidator);
```
Comment (GuySartorelli) [inline src/Context/LoginContext.php:323]: Without this change, we [get this error](https://github.com/creative-commoners/recipe-kitchen-sink/actions/runs/9217532908/job/25363469794):
> You've already used that password in the past, please choose a new password

The comment makes it clear this was the intention here.
Outside of testing I can't see any reason to _not_ have a member password validator, and the current behaviour is when `null` is passed in the result is the default password validator is instantiated which IMO is good behaviour.

File: src/Context/LoginContext.php [hunk 2]
```diff
@@ -315,14 +316,21 @@ protected function generateMember($email, $password, $group, $identifier)
...
+
+        // Write member record
         $member->FirstName = $identifier;
         $member->Surname = "User";
         $member->Email = $email;
-        $member->PasswordEncryption = "none";
```
Comment (GuySartorelli) [inline src/Context/LoginContext.php:322]: This is the main change for this PR - anything else is done to support this change.

**Review by emteknetnz — APPROVED** [review]

**Pull request merged**